core/controllers/classroom.py

```diff
@@ -359,3 +377,136 @@ def get(self, classroom_url_fragment: str) -> None:
+    """Creates a new classroom."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'name': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'has_length_at_most',
+                        'max_value': constants.MAX_CHARS_IN_CLASSROOM_NAME
+                    }, {
+                        'id': 'is_nonempty',
+                    }]
+                }
+            },
+            'url_fragment': constants.SCHEMA_FOR_CLASSROOM_URL_FRAGMENTS,
+        }
+    }
+
+    @acl_decorators.open_access
+    def post(self) -> None:
+        """Creates a new classroom.
+
+        Raise:
+            InvalidInputException. If there are validation errors
+                with name or url_fragment.
+        """
+        assert self.normalized_payload is not None
+
+        name = self.normalized_payload['name']
+        url_fragment = self.normalized_payload['url_fragment']
+
+        new_classroom_id = classroom_config_services.get_new_classroom_id()
+        classroom_config_services.create_new_default_classroom(
+            new_classroom_id, name, url_fragment
+        )
+
+        self.render_json({
+            'new_classroom_id': new_classroom_id
+        })
+
+
+class AllTopicsClassroomInfoHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """return a list of all topics and their
+    relation with classrooms.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
+    HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {'GET': {}}
+
+    @acl_decorators.open_access
+    def get(self) -> None:
+        topic_dicts = [
+            topic.to_dict() for topic in topic_fetchers.get_all_topics()
+        ]
+
+        classrooms = classroom_config_services.get_all_classrooms()
+        topics_classroom_info_dicts: Dict[
+            str, Dict[str, str|None]] = {}
+
+        for topic_dict in topic_dicts:
+            topics_classroom_info_dicts[topic_dict['id']] = {
+                'topic_id': topic_dict['id'],
+                'topic_name': topic_dict['name'],
+                'classroom_name': None,
+                'classroom_url_fragment': None
+            }
+
+        for classroom in classrooms:
+            for topic_id in classroom.get_topic_ids():
+                topics_classroom_info_dicts[topic_id].update({
+                    'classroom_name': classroom.name,
+                    'classroom_url_fragment': classroom.url_fragment
+                })
+
+        self.render_json({
+            'all_topics_classroom_info': list(
+                topics_classroom_info_dicts.values())
+        })
+
+
+class AllClassroomsSummaryHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """return a list of subset of classroom properties.
```

Comment:
subset? That's not a very descriptive word, is there something that semantically binds the returned properties? Return a list of classroom properties needed for a preview?

Reply:
I agree. I have updated it to "return a list of properties which are needed to show a classroom card."